### ISSUE TITLE
Dynamic configuration of DNS server addresses

### ISSUE THREAD

**Issue by chunyingw** [bluejekyll/trust-dns#1016]
Ben,

Do we have a way to update the DNS server addresses for AsyncResolver at runtime ? Ex. for each query, it could use different DNS servers if no record is found in the cache. 

My use case is that the resolver  service could learn DNS server address from different sources such as DHCP client, NDP and default ones. So the DNS server addresses are updated dynamically.

I did a code search and did not find an obvious way to do that or I miss something?

**Comment by bluejekyll** [comment]
No, we currently don't support this. I've wanted to do this, I think it will be needed for things like changing networks etc.

I don't currently have a design for it as I think it's a somewhat complex and possibly OS specific operation. My guess is that we want to have a high-level API that can be called to perform the update, and then each OS can have a wrapper that would update the connections via whatever mechanism it needs to implement...

**Comment by chunyingw** [comment]
Thanks for the response, Ben.

As a temporary workaround for me, whenever the DNS name server address configuration update, a new AsyncResolver instance will be created. Meanwhile I may look into how to make dynamic configuration of DNS name servers work for trust-dns-resolver.
